File: 设计模式/PublishSubscriber.js
class Publisher {
  constructor() {
    this.subscriber = [];
  }

  add(sub) {
    this.subscriber.push(sub);
  }

  notify(mas) {
    this.subscriber.map((item) => {
      item.update(mas + "中国崛起了");
    });
  }
}

class Subscriber {
  constructor(name) {
    this.name = name;
  }

  update(msg) {
    console.log(`${this.name}---${msg}`);
  }
}

const sub1 = new Subscriber("张飞");
const sub2 = new Subscriber("赵云");

const publisher = new Publisher();

publisher.add(sub1);
publisher.add(sub2);

publisher.notify("王巽星");

File: 设计模式/PublishSubscriber.py
class Publish:
    def __init__(self):
        self.subs = []

    def add(self, sub):
        self.subs.append(sub)

    def notify(self, msg):
        for sub in self.subs:
            sub.print(msg)


class Sub:
    def __init__(self, name):
        self.name = name

    def print(self,msg):
          print(f"{self.name}{msg}")


s1=Sub("zhangsan")
s2=Sub("lisi")

p=Publish()

p.add(s1)
p.add(s2)

p.notify("zuishuai")